mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala

```diff
@@ -155,24 +182,47 @@ class CountVectorizer @Since("1.5.0") (@Since("1.5.0") override val uid: String)
     transformSchema(dataset.schema, logging = true)
     val vocSize = $(vocabSize)
     val input = dataset.select($(inputCol)).rdd.map(_.getAs[Seq[String]](0))
+    val filteringRequired = isSet(minDF) || isSet(maxDF)
```

Comment:
this would cause a count also when it is not necessary, ie. when `minDF`/`maxDF` are set not as a fraction but as an integer

Reply:
I imagine you just want the same conditions that are used below, like `$(minDF) >= 1.0`

Reply:
yes, this is the right condition to choose whether  to `filter` or not, but the right one to choose whether to do the count or not is the one you just stated

Reply:
Changed.